### DISCUSSION TITLE
Data migration from Opensearch to Operator managed Opensearch

### DISCUSSION THREAD

**Discussion by IshaGirdhar** [Opster/opensearch-k8s-operator#422] [Q&A] [answered]
What is the best approach to migrate data in production when we upgrade from our standalone Opensearch to Operator?

**Comment by swoehrl-mw** [comment, marked as answer]
Hi @IshaGirdhar. I suggest you use the snapshot feature of opensearch to snapshot the indices on your old cluster and then restore the snapshot to the new cluster.
Ideally you have an existing backup process in place and can just use that and point the restore to the new cluster.
If you first restore older data you can also try to minimize downtime.
You can of course also look for other data migration tools out there. In that regard any cluster created by the operator is just a normal opensearch cluster.

**Reply by IshaGirdhar** [reply]
@swoehrl-mw Thanks for your answer. Also, a follow-up question on that is, as we have an already running standalone Opensearch cluster, can the operator manage that instead of creating a new one, in that case, we can avoid the migration of data altogether.

**Reply by swoehrl-mw** [reply]
There is no explicit support for "adopting" clusters into the operator. Theoretically it could work if the statefulsets and volumes are all named as the operator would create them. If you then create a cluster spec with compatible names the operator could update the existing statefulsets. I guess it would probably still lead to downtime as the specs are likely not directly compatible (for example the PodManagementPolicy will be different which means the STS has to be recreated manually).
You can try it, but are on your own with it.
